src/sentry/api/endpoints/group_hashes.py

```diff
@@ -32,15 +37,29 @@ def get(self, request, group):
         :auth: required
         """
 
-        queryset = GroupHash.objects.filter(
-            group=group.id,
+        data_fn = partial(
+            lambda *args, **kwargs: raw_query(*args, **kwargs)['data'],
+            start=datetime.utcfromtimestamp(0),  # will be clamped to project retention
+            end=datetime.utcnow(),  # will be clamped to project retention
+            aggregations=[
+                ('argMax(event_id, timestamp)', None, 'event_id'),
+                ('max', 'timestamp', 'latest_event_timestamp')
+            ],
+            filter_keys={
+                'project_id': [group.project_id],
+                'group_id': [group.id]
+            },
+            groupby=['primary_hash'],
+            referrer='api.group-hashes',
+            orderby=['-latest_event_timestamp'],
         )
 
+        handle_results = partial(self.__handle_results, group.project_id, group.id, request.user)
+
         return self.paginate(
             request=request,
-            queryset=queryset,
-            order_by='id',
-            on_results=lambda x: serialize(x, request.user),
+            on_results=lambda results: handle_results(results),
```

Comment:
```suggestion
            on_results=handle_results,
```

Since this is partialed already, there's not really any benefit to wrapping it in a `lambda`.